invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js

```diff
@@ -5,32 +5,136 @@
+  const [formats, setFormats] = useState([]);
+
+  useEffect(async () => {
+    try {
+      const response = await http.get(
+        '/api/oaipmh/formats'
+      );
+      const formats = response.data?.hits?.hits;
+      if (Array.isArray(formats) && formats.length > 0) {
+        const serialized = formats.map((formt) => {
+          return {
+            key: formt.id,
+            value: formt.id,
+            text: knownFormats[formt.id] ?? formatKeyToName(formt.id)
+          }
+        });
+        setFormats(serialized);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
+
+  /**
+   * Replaces the metadata prefix in the link.
+   * Used to update links when the metadata prefix is changed.
+   */
+  const replaceLinkPrefix = (link, newPrefix) => {
+    const oldPrefix = getPrefixFromLink(link);
+    if (oldPrefix) {
+      return link.replace(oldPrefix, newPrefix)
+    }
+    return link;
+  };
+
   /**
-   * WIP: this method should remove metadataPrefix from the url once the attribute is listed
-   * as a dropdown.
+   * Retrieves metadata prefix from a link.
+   * Uses a regex to retrieve the prefix, matching "&metadataPrefix=...&".
    *
-   * @param {string} url
-   * @returns {string} stripped url
+   * Returns null if pattern not found.
    */
-  const sanitizeLink = (url) => {
-    return url;
+  const getPrefixFromLink = (link) => {
+    const prefixRegex = /\&metadataPrefix\=(.*)\&/;
+    const match = link.match(prefixRegex);
+    if (match && match.length) {
+      const prefix = match[1];
+      return prefix;
+    }
+    return null;
   };
 
-  const { links } = data;
-  const listRecords = sanitizeLink(links["oai-listrecords"]);
-  const listIdentifiers = sanitizeLink(links["oai-listidentifiers"]);
+  /**
+   * Transforms a key into a readable name.
+   * @example
+   *  // returns Oai Datacite
+   *  formateKeytoName('oai_datacite')
+   */
+  const formatKeyToName = (formatKey) => {
+    const whiteSpaced = formatKey.replace("_", " ");
+    const capitalised = capitalize(whiteSpaced);
+    return capitalised;
+  };
+
+  /**
+   * Dropdown's on change handler.
+   * This method will update the following states if the new prefix is known:
+   * - 'links', each link with its prefix replaced by the new prefix.
+   */
+  const prefixOnChange = (event, data) => {
+    const newPrefix = data.value;
+    if (Object.values(formats).includes(newPrefix)) {
+      const newLinks = {};
+      Object.keys(links).map((key) => {
+        let link = links[key];
+        let newLink = replaceLinkPrefix(link, newPrefix);
+        newLinks[key] = newLink;
+      });
+      setLinks(newLinks);
+    }
+  };
+
+  const listRecords = links["oai-listrecords"];
+  const listIdentifiers = links["oai-listidentifiers"];
+
+  // Set default prefix based on backend link, use dublin core if unknown.
+  let defaultPrefix = getPrefixFromLink(listRecords)
+  if (!Object.prototype.hasOwnProperty.call(formats, defaultPrefix)) {
```

Comment:
The dropdown will have as a default the format currently in use (e.g. the one retrieved from the backend).
However, if that format is not known, it will use OAI DublinCore by default.

Reply:
I might lack some context about this, but I think that we shouldn't change forcefully the format to simply adapt it to "our" frontend needs. This is connected to the previous comment, I think it can be addressed in a separate issue.